File: frontend/lib/models/tmp_models/message_model.dart
import 'package:flutter/foundation.dart';
import 'package:freezed_annotation/freezed_annotation.dart';
import 'package:tinder/models/tmp_models/attachment_model.dart';

part 'message_model.freezed.dart';

part 'message_model.g.dart';

@freezed
class MessageModel with _$MessageModel {
  @JsonSerializable(fieldRename: FieldRename.snake, explicitToJson: true)
  const factory MessageModel({
    int? id,
    List<AttachmentModel>? attachments,
    MessageSender? messageSender,
    String? createdAt,
    String? updatedAt,
    String? readAt
  }) = _MessageModel;

  factory MessageModel.fromJson(Map<String, dynamic> json) =>
      _$MessageModelFromJson(json);
}

enum MessageSender { a, b }

File: backend/src/models/message_model.ts
enum MessageSender { a, b }

interface MessageModel {
    id?: number,
    attachments: AttachmentModel[],
    created_at?: string,
    updated_at?: string,
    read_at?: string,
    message_sender?: string
}

interface MessageModelService {
    id?: string,
    attachments: AttachmentModel[],
    createdAt?: number,
    updatedAt?: number,
    readAt?: number,
    messageSender?: MessageSender
}

interface MessageTableElement {
    id: number,
    created_at: string,
    updated_at?: string,
    read_at?: string
    message_sender_id: number
}